pkg/template/controller/readiness.go

```diff
@@ -150,13 +148,21 @@ var readinessCheckers = map[schema.GroupKind]func(client.Interface, runtime.Obje
 
 // canCheckReadiness indicates whether a readiness check exists for a GK.
 func canCheckReadiness(ref kapi.ObjectReference) bool {
+	if ref.GroupVersionKind().GroupKind() == buildapi.LegacyKind("BuildConfig") ||
+		ref.GroupVersionKind().GroupKind() == buildapi.Kind("BuildConfig") {
+		return true
+	}
 	_, found := readinessCheckers[ref.GroupVersionKind().GroupKind()]
 	return found
 }
 
 // checkReadiness runs the readiness check on a given object.  TODO: remove
 // "oc client.Interface" and error once BuildConfigs can report on the status of
 // their latest build.
-func checkReadiness(oc client.Interface, ref kapi.ObjectReference, obj runtime.Object) (bool, bool, error) {
-	return readinessCheckers[ref.GroupVersionKind().GroupKind()](oc, obj)
+func checkReadiness(oc buildclient.Interface, ref kapi.ObjectReference, obj runtime.Object) (bool, bool, error) {
```

Comment:
This method plumbs through correctly?  Also, we should have have an issue.  This is the only one live checking.  it should stop.

Reply:
@bparees